docs/extend/component-registries.md

```diff
@@ -0,0 +1,240 @@
+| Attribute | Type | Input or Output | Actions | Description |
+| `ZWE_CLI_REGISTRY_COMMAND` | string | Input | All | Values of 'install', 'upgrade', 'uninstall', 'search' inform handler which action to take and what additional input & output to expect |
+| `ZWE_CLI_PARAMETER_REGISTRY` | string | Input | Install, Upgrade, Uninstall, Search | Used to inform handler which registry to use. Can be any format the handler understands. |
+| `ZWE_CLI_REGISTRY_DRY_RUN` | boolean | Input | Install, Upgrade, Uninstall | If true, handler should show as much as possible about what would happen during this command, without committing changes that would alter which components are installed. |
+| `ZWE_CLI_PARAMETER_COMPONENT_NAME` | string | input | Install, Upgrade, Uninstall, Search | Value varies by command. For 'install' and 'uninstall', this value is the exact name of a component. For upgrade, it may also be 'all' to perform an upgrade for all components possible. For 'search', it may be any string to perform searching for exact or partial matching component names. |
+| `ZWE_CLI_PARAMETER_COMPONENT_FILE` | string | output | Install, Upgrade, Uninstall | A comma-separated list of components that have been added or removed. During 'install' or 'upgrade', the list must be full unix paths to component folders or archives that were added. For 'uninstall', the list must instead be just the names of the components that were removed. If the handler failed during its operation or there were no changes, the output should instead just be the string 'null'. |
+
+An example of running `zwe components install -o exact-component-name --handler npm --registry https://zowe` would have the handler being given the following environment variables:
+
+```
+ZWE_CLI_REGISTRY_COMMAND=install
+ZWE_CLI_PARAMETER_REGISTRY=https://zowe
+ZWE_CLI_REGISTRY_DRY_RUN=false
+ZWE_CLI_PARAMETER_COMPONENT_NAME=exact-component-name
+```
+
+And after the command completes, the handler can print anything in STDOUT and STDERR as long as STDOUT includes a line specifying the location of the components installed, via `ZWE_CLI_PARAMETER_COMPONENT_FILE`. The output could look like:
+
+```
+2 packages installed
+Operation successful
+return code=0
+ZWE_CLI_PARAMETER_COMPONENT_FILE=/my/components/exact-component-name/archive.pax,/my/components/dependency1
+```
+
+Where `archive.pax` is an archive of `exact-component-name`, while 'dependency1' is a folder containing the un-archived contents of `dependency1`.
+
+
+### Handler code
+
+The Handler API interface is [located within Zowe's code here](https://github.com/zowe/zowe-install-packaging/blob/2751a194048f0050fc7ebcaeaac8c96a36106991/bin/commands/components/handlerutils.ts)
+
+And Zowe delivers a handler written for use with [npm, located here](https://github.com/zowe/zowe-install-packaging/blob/2751a194048f0050fc7ebcaeaac8c96a36106991/bin/commands/components/npm.ts)
+
+
+## Packaging & Distributing components
```

Comment:
Shouldn't this be titled like, 
Packaging Manager Requirements? or Packaging Manager Objectives